lib/fast_jsonapi/serialization_core.rb

```diff
@@ -26,9 +26,17 @@ def ids_hash(ids, record_type)
         id_hash(ids, record_type) # ids variable is just a single id here
       end
 
-      def attributes_hash(record)
-        attributes_to_serialize.each_with_object({}) do |(key, method_name), attr_hash|
-          attr_hash[key] = record.public_send(method_name)
+      def attributes_hash(record, fields = nil)
```

Comment:
Sparse fieldsets are supposed to work with attributes and relationships as per the jsonapi spec. Here is the definition of fields http://jsonapi.org/format/#document-resource-object-fields

I think we would have to only return the relationships listed in the fieldsets. We would need to add changes for that

Reply:
If I understand what you mean, this might conflict with `include` option. Suppose you're returning main model A and models B and C in relationships. And you want sparse fieldsets in A and B, but C must be full. So if we were to treat keys from `fields` as explicit include, we would have to enumerate all fields there. But if we only treat fields as a guide of which fields to render per model, then we're fine if we don't have fieldset for some model, but it's included (this is how it works now).

Reply:
@youroff I am ok with having the implicit assumption that if a model is not mentioned in the fieldsets then we its included.

I was actually asking about relationships. Let's take the example you have mentioned above. I will simplify it a bit.
```ruby
options = {}
options[:fields] = {
  movie: ['name'], # will only include name attribute into attributes hash
}
MovieSerializer.new([movie, movie], options).serialized_json
```
Let's assume this is our movie serializer
```ruby
class MovieSerializer
    include FastJsonapi::ObjectSerializer
    attributes :name, :release_year
    belongs_to :movie_type
end
```

the output of the serializer would it have the movie_type relationship in the json? or would it only have the name attribute. If I understand the specification correctly it shouldn't have any relationships just the name attribute. However i dont think this implementation is taking care of that scenario.

Reply:
Ah... Gotcha. Yeah, would make sense if relationships were `fields`-aware too. I'll fix that.